src/pyscipopt/expr.pxi

```diff
@@ -109,13 +109,7 @@ class Term:
     def __repr__(self):
         return 'Term(%s)' % ', '.join([str(v) for v in self.vartuple])
 
-    def _evaluate(self, point):
-        '''computes the value of the term in the given variable-value mapping.'''
-        prod = 1
-        for var in self.vartuple:
-            prod *= point[var]
-        return prod
-        
+
```

Comment:
I've have taken the liberty to simplify the implementation of how `Expr` are evaluated. Before there was some recursion that was a bit confusing and I think that now it is more clear with everything being handled under `Solution`.